src/main/java/de/retest/recheck/report/TestReportFilter.java

```diff
@@ -0,0 +1,133 @@
+package de.retest.recheck.report;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import de.retest.recheck.ignore.Filter;
+import de.retest.recheck.report.action.ActionReplayData;
+import de.retest.recheck.ui.actions.ExceptionWrapper;
+import de.retest.recheck.ui.actions.TargetNotFoundException;
+import de.retest.recheck.ui.descriptors.Element;
+import de.retest.recheck.ui.descriptors.SutState;
+import de.retest.recheck.ui.diff.AttributesDifference;
+import de.retest.recheck.ui.diff.ElementDifference;
+import de.retest.recheck.ui.diff.IdentifyingAttributesDifference;
+import de.retest.recheck.ui.diff.LeafDifference;
+import de.retest.recheck.ui.diff.RootElementDifference;
+import de.retest.recheck.ui.diff.StateDifference;
+
+public class TestReportFilter {
+
+	private TestReportFilter() {
+	}
+
+	public static TestReport filter( final TestReport report, final Filter filter ) {
+		final TestReport newTestReport = new TestReport();
+		for ( final SuiteReplayResult suiteReplayResult : report.getSuiteReplayResults() ) {
+			newTestReport.addSuite( TestReportFilter.filter( suiteReplayResult, filter ) );
```

Comment:
Sometimes you do `TestReportFilter.filter( ... )` and sometimes only `filter( ... )`. Just stick to one style (personally, I prefer the latter).